File: editor/en/[287]Find the Duplicate Number.java
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int findDuplicate(int[] nums) {
        boolean[] isDupicated = new boolean[nums.length];

        for (int num : nums) {
            if (isDupicated[num]) {
                return num;
            }

            isDupicated[num] = true;
        }

        return -1;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[287]Find the Duplicate Number.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        isDuplicate = [False] * len(nums)

        for num in nums:
            if isDuplicate[num]:
                return num

            isDuplicate[num] = True

        return -1
# leetcode submit region end(Prohibit modification and deletion)
